### PR TITLE
4.6.4.0.0

### PR DESCRIPTION
**Pull request by helium-service**
This PR updates the partner version to 4.6.4.0.0 for release.

### CODE REVIEW

File: HyprMXAdapter/src/main/java/com/chartboost/mediation/hyprmxadapter/HyprMXAdapter.kt [hunk 1]
```diff
@@ -756,98 +747,43 @@ class HyprMXAdapter : PartnerAdapter {
...
         }
 
-        override fun onAdClosed(
-            placement: Placement?,
-            finished: Boolean,
-        ) {
+        override fun onAdClosed(placement: Placement, finished: Boolean) {
             PartnerLogController.log(DID_DISMISS)
-            listener.onPartnerAdDismissed(
+            request.adInteractionListener.onDismissed(
```
Comment (bichenwang) [inline HyprMXAdapter/src/main/java/com/chartboost/mediation/hyprmxadapter/HyprMXAdapter.kt:786]: unfortunately, this won't build for 4.0. This was added to the partner load request for adaptive banners and is not available until 4.6 or so.
you're going to have to store the original partner ad listener somewhere and pass it to each load somehow. We typically have been storing them in maps keyed by request ID, but it's gotta live somewhere.

Comment (bwised) [reply HyprMXAdapter/src/main/java/com/chartboost/mediation/hyprmxadapter/HyprMXAdapter.kt:786]: Ok, makes sense.  And I found an example to replicate.

File: HyprMXAdapter/src/main/java/com/chartboost/mediation/hyprmxadapter/HyprMXAdapter.kt [hunk 2]
```diff
@@ -646,7 +593,10 @@ class HyprMXAdapter : PartnerAdapter {
      *
      * @return Result.success(PartnerAd) if the ad was successfully shown, Result.failure(Exception) otherwise.
      */
-    private suspend fun showFullscreenAd(partnerAd: PartnerAd): Result<PartnerAd> {
+    private suspend fun showFullscreenAd(
+        partnerAd: PartnerAd,
+        listener: PartnerAdListener?,
```
Comment (chauduyphanvu) [inline HyprMXAdapter/src/main/java/com/chartboost/mediation/hyprmxadapter/HyprMXAdapter.kt:598]: Need docs for `listener`

File: HyprMXAdapter/src/main/java/com/chartboost/mediation/hyprmxadapter/HyprMXAdapter.kt [hunk 3]
```diff
@@ -623,18 +568,20 @@ class HyprMXAdapter : PartnerAdapter {
      */
     private suspend fun loadRewardedAd(
         request: PartnerAdLoadRequest,
-        partnerAdListener: PartnerAdListener,
+        listener: PartnerAdListener,
```
Comment (chauduyphanvu) [inline HyprMXAdapter/src/main/java/com/chartboost/mediation/hyprmxadapter/HyprMXAdapter.kt:571]: Need docs for `listener`

**Review by chauduyphanvu — APPROVED** [review]
lgtm otherwise

File: HyprMXAdapter/src/main/java/com/chartboost/mediation/hyprmxadapter/HyprMXAdapter.kt [hunk 4]
```diff
@@ -857,48 +753,16 @@ class HyprMXAdapter : PartnerAdapter {
             )
         }
 
-        override fun onAdDisplayError(
-            placement: Placement?,
-            hyprMXError: HyprMXErrors?,
-        ) {
+        override fun onAdDisplayError(placement: Placement, hyprMXError: HyprMXErrors) {
```
Comment (bichenwang) [inline HyprMXAdapter/src/main/java/com/chartboost/mediation/hyprmxadapter/HyprMXAdapter.kt:756]: did they move to a kotlin sdk or something? that's why we're able to move to not `?`?

**Review by bichenwang — APPROVED** [review]
nice! good to see hyprmx follow modern sdk design, too xD

**Review by CB-RyanMcCormick — APPROVED** [review]

File: HyprMXAdapter/src/main/java/com/chartboost/mediation/hyprmxadapter/HyprMXAdapter.kt [hunk 5]
```diff
@@ -857,48 +795,33 @@ class HyprMXAdapter : PartnerAdapter {
...
 
-        override fun onAdRewarded(
-            placement: Placement,
-            rewardName: String,
-            rewardValue: Int,
-        ) {
+    private class ShowRewardedAdListener(
+        private val request: PartnerAdLoadRequest,
+        private val listener: PartnerAdListener?,
+    ) : ShowAdListener(request, listener), HyprMXRewardedShowListener {
```
Comment (bwised) [inline HyprMXAdapter/src/main/java/com/chartboost/mediation/hyprmxadapter/HyprMXAdapter.kt:821]: This was added due to the issue https://chartboost.atlassian.net/browse/HB-7718 during certification.

File: HyprMXAdapter/src/main/java/com/chartboost/mediation/hyprmxadapter/HyprMXAdapter.kt [hunk 6]
```diff
@@ -476,61 +476,60 @@ class HyprMXAdapter : PartnerAdapter {
...
-                loadAd(
-                    listener =
-                    object : HyprMXLoadAdListener {
-                        override fun onAdLoaded(isAdAvailable: Boolean) {
-                            if (isAdAvailable) {
-                                PartnerLogController.log(LOAD_SUCCEEDED)
-                                continuation.resume(
-                                    Result.success(
-                                        PartnerAd(
-                                            ad = this,
```
Comment (bwised) [inline HyprMXAdapter/src/main/java/com/chartboost/mediation/hyprmxadapter/HyprMXAdapter.kt]: `this` was bad.

File: HyprMXAdapter/src/main/java/com/chartboost/mediation/hyprmxadapter/HyprMXAdapter.kt [hunk 7]
```diff
@@ -476,61 +476,60 @@ class HyprMXAdapter : PartnerAdapter {
...
+            banner.loadAd(
+                listener =
+                object : HyprMXLoadAdListener {
+                    override fun onAdLoaded(isAdAvailable: Boolean) {
+                        if (isAdAvailable) {
+                            PartnerLogController.log(LOAD_SUCCEEDED)
+                            continuation.resume(
+                                Result.success(
+                                    PartnerAd(
+                                        ad = banner,
```
Comment (bwised) [inline HyprMXAdapter/src/main/java/com/chartboost/mediation/hyprmxadapter/HyprMXAdapter.kt:516]: this is good.

**Pull request merged**